packages/swap/src/utils/getPoolQuote.ts

```diff
@@ -0,0 +1,81 @@
+import { bigintMin } from '@/shared/functions';
+import { QuoteQueryResponse, SwapFee } from '@/shared/schemas';
+import {
+  buildFee,
+  calculateIncludedSwapFees,
+  estimateIngressEgressFeeAssetAmount,
+} from '@/swap/utils/fees';
+import { estimateSwapDuration } from '@/swap/utils/swap';
+import logger from './logger';
+import { getMinimumEgressAmount, getEgressFee } from './rpc';
+import ServiceError from './ServiceError';
+import { getSwapRate } from './statechain';
+import { InternalAsset } from '../client';
+import { checkPriceWarning } from '../pricing/checkPriceWarning';
+import { QuoteType } from '../quoting/Quoter';
+
+export default async function getPoolQuote(
+  srcAsset: InternalAsset,
+  destAsset: InternalAsset,
+  swapInputAmount: bigint,
+  ingressEgressFeeIsGasAssetAmount: boolean,
+  fees: SwapFee[],
+): Promise<QuoteQueryResponse & { quoteType: QuoteType }> {
+  const includedFees: SwapFee[] = [...fees];
+
+  const start = performance.now();
+
+  const quote = await getSwapRate({ srcAsset, destAsset, amount: swapInputAmount });
+
+  const lowLiquidityWarning = await checkPriceWarning({
+    srcAsset,
+    destAsset,
+    srcAmount: swapInputAmount,
+    destAmount: BigInt(quote.outputAmount),
+  });
+
+  const quoteSwapFees = await calculateIncludedSwapFees(
+    srcAsset,
+    destAsset,
+    swapInputAmount,
+    quote.intermediateAmount,
+    quote.outputAmount,
+  );
+  includedFees.push(...quoteSwapFees);
+
+  let egressFee = await getEgressFee(destAsset);
+  if (egressFee == null) {
+    throw ServiceError.internalError(`could not determine egress fee for ${destAsset}`);
+  }
+  if (ingressEgressFeeIsGasAssetAmount) {
+    egressFee = await estimateIngressEgressFeeAssetAmount(egressFee, destAsset);
+  }
```

Comment:
I'm not entirely sure but this looks to me like its still doing back comp. for <1.3.
If it is, I think we can safely remove that check at this point ?

Reply:
We can, but it changes a lot of tests. Tried once already, it makes the PR even harder to review